### PR TITLE
Display user profile image on overhang

### PR DESCRIPTION
**Pull request by thomabx**
If there is a user profile for the currently logged in user, add the profile image next to the user name on the overhang.

### CODE REVIEW

**Review by neilsb — APPROVED** [review]

File: source/Main.brs [hunk 1]
```diff
@@ -25,6 +25,12 @@ sub Main()
   ' load home page
   m.overhang.title = "Home"
   m.overhang.currentUser = m.user.Name
+  if m.user.PrimaryImageTag <> invalid then
+    userImage = UserImageURL(m.user.id, { "tag": m.user.PrimaryImageTag })
```
Comment (cewert) [inline source/Main.brs:29]: By default, `UserImageURL()` grabs a 300px by 300px image and then we are forcing that image into a 64px by 64px poster.

We should set `maxHeight `and `maxWidth `to 64:
`userImage = UserImageURL(m.user.id, { "tag": m.user.PrimaryImageTag, "maxHeight": 64, "maxWidth": 64 })`

**Review by cewert — CHANGES_REQUESTED** [review]

File: components/JFOverhang.xml [hunk 2]
```diff
@@ -22,7 +22,9 @@
       <Label id="overlayTitle" font="font:LargeSystemFont" vertAlign="center" height="64" width="1100" />
```
Comment (cewert) [inline components/JFOverhang.xml:22]: The width of overlayTitle should be adjusted when displaying the user profile image. If not, they will overlap when displaying media with a long title

**Review by cewert — CHANGES_REQUESTED** [review]

**Review by cewert — COMMENTED** [review]
Overall this looks good although I wonder if this is really needed and if there will be any performance hit.

By adding the user image we take away space for media with long titles and increase the chance of the title being cut off. The extra API call means a longer wait till the first two rows are rendered on startup. The performance hit should be minimal but for people with remote servers and/or crappy rokus it may have an impact

**Comment by thomabx** [comment]
> Overall this looks good although I wonder if this is really needed and if there will be any performance hit.

I wouldn't say it's needed, but it does match the web client behavior of displaying the profile image.

> By adding the user image we take away space for media with long titles and increase the chance of the title being cut off. The extra API call means a longer wait till the first two rows are rendered on startup. The performance hit should be minimal but for people with remote servers and/or crappy rokus it may have an impact

We can add a task to load the image which would rely on the bandwidth to the server but allow rendering. We could additionally make it an option that defaults to off. What do you think?

**Comment by bisby** [comment]
> We can add a task to load the image which would rely on the bandwidth to the server but allow rendering

I'll admit I've been absent lately and not following every PR closely... But currently,
for design simplicity, all the API requests are done synchronously. (ie, wait for the response
as the request is made, rather than letting it handle in the background).

This sounds like a great opportunity to actually start shifting towards properly
making async requests. Request the image, draw a default or empty box, load image
as it becomes available.

:man_shrugging:

**Comment by cewert** [comment]
> I wouldn't say it's needed, but it does match the web client behavior of displaying the profile image.

Does it? The web client is showing a rounded image with no text for me right now(10.5.3) vs the square image + text in this PR.

> We can add a task to load the image which would rely on the bandwidth to the server but allow rendering. We could additionally make it an option that defaults to off. What do you think?

Loading the image using a task or async API call is a great idea as well as an option set to off by default but is it worth all the effort? How many people are really going to turn this feature on? (I wouldn't because of the drawbacks I've listed)

The web client is great and I agree we should be looking to it for direction but at the same time the roku app will never be the web client. The computers running the apps are different, the way you interact with the app is different (remote vs mouse), the UI is different(one is meant to be viewed from 2 feet away the other is meant to be viewed 6+ feet away) etc. etc. At some point we have to make our own path. For example, the web client doesn't show the time does that mean we should remove it from our app?

**Pull request closed without merging**